### PR TITLE
fix(router-core): reject overlapping parameter affixes

### PR DESCRIPTION
**Pull request by Sheraff**
## Summary

- reject route matches when a parameter prefix and suffix overlap within the matched path
- apply the length guard to required, optional, and wildcard parameters
- preserve valid empty parameters where the affixes meet without overlapping
- add regression coverage for all three parameter kinds

## Context

A route such as `/ab{$id}bc` incorrectly matched `/abc`: `abc` starts with `ab` and ends with `bc`, but both affixes claimed the same `b`. Parameter extraction then returned that shared character as the parameter value.

The matcher now requires the matched path to be at least as long as the combined prefix and suffix before accepting both affixes. `/abc` is rejected, while `/abbc` remains valid with an empty parameter value.

## Bundle size

`react-router.minimal`: +4 B gzip locally.

## Bug Fixes

* Improved route matching for dynamic, optional, and wildcard segments with overlapping prefixes and suffixes.
* Prevented invalid or incomplete URLs—including paths with extra trailing characters—from matching routes incorrectly.
* Preserved complete wildcard path remainders during suffix matching.
* Ensured route parameters are extracted correctly when prefixes or suffixes are absent.

## Tests

* Added coverage for valid and invalid routes involving overlapping prefixes and suffixes, including trailing-character edge cases.

### CODE REVIEW

File: packages/router-core/tests/new-process-route-tree.test.ts [hunk 1]
```diff
@@ -686,6 +686,14 @@ describe('findRouteMatch', () => {
...
         )
       })
+      it.each(['/ab{$id}bc', '/ab{-$id}bc', '/ab{$}bc'])(
+        'does not match overlapping affixes for %s',
+        (route) => {
+          const tree = makeTree([route])
+          expect(findRouteMatch('/abc', tree)).toBeNull()
+          expect(findRouteMatch('/abbc', tree)?.route.id).toBe(route)
+        },
+      )
```
[the remark this answers is not in the document]
Comment (Sheraff) [reply packages/router-core/tests/new-process-route-tree.test.ts]: Addressed in bdd2855925. The accepted cases now assert rawParams for required, optional, and wildcard parameters, including a non-empty multi-segment wildcard remainder.
